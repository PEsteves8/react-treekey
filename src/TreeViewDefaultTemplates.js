import React from "react";
import styles from "./defaultTreeStyles";

export let defaultToggle = (isExpanded) => {
  const svgStyles = {
    fill: "white",
    strokeWidth: 0,
    transformOrigin: "center",
    transform: `rotate(${isExpanded ? "90deg" : "0"})`,
  };

  let { height, width, points } = styles.nodeToggleSvg;

  return (
    <svg height={height} width={width}>
      <polygon points={points} style={svgStyles} />
    </svg>
  );
};

export let defaultHeader = (node) => {
  return <span style={{ paddingLeft: "7px" }}>{node.name}</span>;
};
