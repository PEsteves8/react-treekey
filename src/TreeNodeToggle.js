import React from 'react';

import styles from './defaultTreeStyles';
import PropTypes from 'prop-types';
import { defaultToggle } from './TreeViewDefaultTemplates';

class TreeNodeToggle extends React.Component {

    constructor(props) {
        super(props);
        this.onClick = this.onClick.bind(this);
    }

    onClick(e) {
        e.stopPropagation();
        this.props.setToggling(this.props.node);
    }

    renderToggle() {
        let { template, isExpanded } = this.props;
        return template ? template(isExpanded) : defaultToggle(isExpanded);
    }

    render() {
        return <div onClick={this.onClick} style={styles.nodeToggleWrapper}>
                    {this.renderToggle()}
                </div>
     }
}

TreeNodeToggle.propTypes = {
    template: PropTypes.func,
    setToggling: PropTypes.func,
    node: PropTypes.object,
    isExpanded: PropTypes.bool,
};

export default TreeNodeToggle;