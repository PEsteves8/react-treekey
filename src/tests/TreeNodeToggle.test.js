import React from "react";
import TreeNodeToggle from "../TreeNodeToggle";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import * as templates from "../TreeViewDefaultTemplates";

const mockNode = {
  name: "some node",
};
const isExpanded = true;
const expandedNodes = [];

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("<TreeNodeToggle>", () => {
  test("should match snapshot with correct template", () => {
    let template = jest.fn();
    template.mockImplementation(() => <span>some node</span>);

    act(() => {
      render(
        <TreeNodeToggle
          template={template}
          isExpanded={isExpanded}
          node={mockNode}
          expandedNodes={expandedNodes}
        ></TreeNodeToggle>,
        container
      );
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(template).toHaveBeenCalledTimes(1);
    expect(template).toHaveBeenCalledWith(isExpanded);
  });

  test("calls default toggle if no template is given", () => {
    const spy = jest.spyOn(templates, "defaultToggle");

    act(() => {
      render(
        <TreeNodeToggle
          node={mockNode}
          isExpanded={isExpanded}
          expandedNodes={expandedNodes}
        ></TreeNodeToggle>,
        container
      );
    });

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(isExpanded);
    spy.mockRestore();
  });

  test("calls props.setToggling when clicked", () => {
    const setToggling = jest.fn();

    act(() => {
      render(
        <TreeNodeToggle
          node={mockNode}
          setToggling={setToggling}
          expandedNodes={expandedNodes}
        ></TreeNodeToggle>,
        container
      );
    });

    const el = container.querySelector("div");

    act(() => {
      el.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(setToggling).toHaveBeenCalledTimes(1);
  });
});
