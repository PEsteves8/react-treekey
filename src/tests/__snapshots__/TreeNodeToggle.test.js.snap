// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeNodeToggle> should match snapshot with correct template 1`] = `"<div style=\\"display: inline-block; padding: 0px 5px; margin-left: -18px;\\"><span>some node</span></div>"`;
